store/src/java-test/com/zimbra/cs/filter/GetFilterRulesTest.java

```diff
@@ -0,0 +1,427 @@
+                + "<filterRules>"
+                  + "<filterRule active=\"1\">"
+                    + "<filterTests>"
+                      + "<headerTest stringComparison=\"matches\" header=\"Subject\" index=\"0\" value=\"*\"/>"
+                    + "</filterTests>"
+                    + "<filterActions>"
+                      + "<actionFileInto folderPath=\"if-block\" index=\"0\" copy=\"0\"/>"
+                    + "</filterActions>"
+                  + "</filterRule>"
+                + "</filterRules>"
+              + "</GetFilterRulesResponse>";
+        XMLDiffChecker.assertXMLEquals(expectedSoapResponse, response.prettyPrint());
+    }
+
+    @Test
+    public void testIfWithAllofAnyOf() throws Exception {
+        // - 'allof' and 'anyof' tests
+        String filterScript
+                    = "require \"tag\";"
+                    + "if allof (header :comparator \"i;ascii-casemap\" :matches \"Subject\" \"*\") {"
+                    + "  fileinto \"if-block1\";"
+                    + "}"
+                    + "if anyof (header :comparator \"i;ascii-casemap\" :matches \"X-Header\" \"*\") {"
+                    + "  fileinto \"if-block2\";"
+                    + "}";
+        Account account = Provisioning.getInstance().getAccount(
+                MockProvisioning.DEFAULT_ACCOUNT_ID);
+        RuleManager.clearCachedRules(account);
+        account.setMailSieveScript(filterScript);
+
+        Element request = new Element.XMLElement(MailConstants.GET_FILTER_RULES_REQUEST);
+        Element response = new GetFilterRules().handle(request, ServiceTestUtil.getRequestContext(account));
+
+        String expectedSoapResponse =
+                "<GetFilterRulesResponse xmlns=\"urn:zimbraMail\">"
+                + "<filterRules>"
+                  + "<filterRule active=\"1\">"
+                    + "<filterTests condition=\"allof\">"
+                      + "<headerTest stringComparison=\"matches\" header=\"Subject\" index=\"0\" value=\"*\"/>"
+                    + "</filterTests>"
+                    + "<filterActions>"
+                      + "<actionFileInto folderPath=\"if-block1\" index=\"0\" copy=\"0\"/>"
+                    + "</filterActions>"
+                  + "</filterRule>"
+                  + "<filterRule active=\"1\">"
+                  + "<filterTests condition=\"anyof\">"
+                    + "<headerTest stringComparison=\"matches\" header=\"X-Header\" index=\"0\" value=\"*\"/>"
+                  + "</filterTests>"
+                  + "<filterActions>"
+                    + "<actionFileInto folderPath=\"if-block2\" index=\"0\" copy=\"0\"/>"
+                  + "</filterActions>"
+                + "</filterRule>"
+                + "</filterRules>"
+              + "</GetFilterRulesResponse>";
+        XMLDiffChecker.assertXMLEquals(expectedSoapResponse, response.prettyPrint());
+    }
+
+    @Test
+    public void testNestedIfWithoutAllof() throws Exception {
```

Comment:
Each test case does call the assertFalse() inside the XMLDiffChecker.assertXMLEquals().